src/gil.rs

```diff
@@ -187,42 +176,65 @@ impl ReleasePool {
     }
 }
 
-static mut POOL: *mut ReleasePool = ::std::ptr::null_mut();
+struct ReleasePool {
+    value: UnsafeCell<Option<ReleasePoolImpl>>,
+}
+
+impl ReleasePool {
+    const fn new() -> Self {
+        Self {
+            value: UnsafeCell::new(None),
+        }
+    }
+    unsafe fn get_or_init(&self) -> &mut ReleasePoolImpl {
+        if (*self.value.get()).is_none() {
+            *self.value.get() = Some(ReleasePoolImpl::new());
+        }
+        (*self.value.get()).as_mut().unwrap()
+    }
+}
+
+static POOL: ReleasePool = ReleasePool::new();
+
+unsafe impl Sync for ReleasePool {}
 
 #[doc(hidden)]
-pub struct GILPool<'p> {
-    py: Python<'p>,
+pub struct GILPool {
     owned: usize,
     borrowed: usize,
     no_send: Unsendable,
 }
 
-impl<'p> GILPool<'p> {
+impl GILPool {
+    /// # Safety
+    /// This function requires that GIL is already acquired.
     #[inline]
-    pub fn new(py: Python) -> GILPool {
+    pub unsafe fn new() -> GILPool {
         increment_gil_count();
-        let p: &'static mut ReleasePool = unsafe { &mut *POOL };
+        let pool = POOL.get_or_init();
```

Comment:
Oh, I think I missed a `release_pointers()` call here:

```suggestion
        // Release objects that were dropped since last GIL acquisition
        let pool = POOL.get_or_init();
        pool.release_pointers();
```

Maybe this is why you removed the second bullet from the CHANGELOG?